code/script.js

```diff
@@ -1,343 +1,463 @@
+  numberOfGuesses++;
+  const category = questions.options[questions.selectedIndex].parentNode.label;
   // This variable stores what option group (category) the question belongs to.
+
   // We also need a variable that stores the actual value of the question we've selected.
+  let optionValue = questions.value;
 
-  if (category === 'hair color') {
+  if (category === "hair color") {
     currentQuestion = {
-      attribute: 'hairColor',
-      // value: ,
+      attribute: "hairColor",
+      value: optionValue,
       // 👆 add the value from the input here
       category: category,
-    }
-  } else if (category === 'eye color') {
-    // Set this up your self
-  } else if (category === 'accessories') {
+    };
+    checkQuestion();
+  } else if (category === "eye color") {
+    currentQuestion = {
+      attribute: "eyeColor",
+      value: optionValue,
+      category: category,
+    };
+    checkQuestion();
+  } else if (category === "accessories") {
     currentQuestion = {
-      //attribute: ,
+      attribute: optionValue,
       // 👆 this is the property of the booleans such as smoke, glasses and hat. add the value from the input here
       value: true, // we're asking if this person wears a hat for exaple, so always true in the question.
       category: category,
-    }
-  } else if (category === 'other') {
+    };
+    checkQuestion();
+  } else if (category === "other") {
+    currentQuestion = {
+      attribute: optionValue,
+      value: true,
+      category: category,
+    };
+    checkQuestion();
     // Set this up your self (should be same structure as above)
   }
-}
+};
 
 // This function should be invoked when you click on 'Find Out'.
 const checkQuestion = () => {
   // Compare the currentQuestion with the secret person.
   // See if we should keep or remove people based on that
   // Then invoke filterCharacters
-}
+  let secretValue = secret[currentQuestion.attribute];
+  if (secretValue === currentQuestion.value) {
+    filterCharacters(true, currentQuestion.category);
+  } else {
+    filterCharacters(false, currentQuestion.category);
+  }
+};
 
 // It'll filter the characters array and redraw the game board.
-const filterCharacters = (keep) => {
+const filterCharacters = (keep, category) => {
   // Show the correct alert message for different categories
-  if (group === 'accessories') {
+  if (category === 'hair color') {
     if (keep) {
-      alert(
-        `Yes, the person wears ${attribute}! Keep all that wears ${attribute}`
-      )
+      alert(`Yes, the secret person has ${currentQuestion.value} hair! The game will now remove everyone that doesn't have ${currentQuestion.value} hair!`);
     } else {
-      alert(
-        `No, the person doesn't wear ${attribute}! Remove all that wears ${attribute}`
-      )
+      alert(`No, the secreet person does not have ${currentQuestion.value} hair! The game will now remove everyone that has ${currentQuestion.value} hair!`);
```

Comment:
Typo ('secreet') 😉

Reply:
and a secret from me: https://marketplace.visualstudio.com/items?itemName=streetsidesoftware.code-spell-checker 😸